### PR TITLE
feat(connector): [Adyen] Add support for PayPal wallet mandates

### PR DESCRIPTION
**Pull request by prasunna09**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Add PayPal wallet mandates in Adyen connector

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

CIT - 
Screen Shot 2023-07-12 at 12 48 59 PM

Screen Shot 2023-07-12 at 12 49 37 PM

Screen Shot 2023-07-12 at 12 50 41 PM

MIT - 
Screen Shot 2023-07-12 at 12 53 12 PM

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/connector/adyen/transformers.rs [hunk 1]
```diff
@@ -1054,6 +1054,21 @@ impl<'a> TryFrom<&api::Card> for AdyenPaymentMethod<'a> {
...
 
+impl TryFrom<storage_enums::PaymentMethodType> for PaymentType {
+    type Error = Error;
+    fn try_from(item: storage_enums::PaymentMethodType) -> Result<Self, Self::Error> {
+        match item {
+            storage_enums::PaymentMethodType::Credit | storage_enums::PaymentMethodType::Debit => {
+                Ok(Self::Scheme)
+            }
+            storage_enums::PaymentMethodType::Paypal => Ok(Self::Paypal),
+            _ => Err(errors::ConnectorError::NotImplemented(
```
Comment (ArjunKarthik) [inline crates/router/src/connector/adyen/transformers.rs:1065]: Doesadyen support mandates only in case of cards and PayPal?

**Review by ArjunKarthik — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/adyen/transformers.rs [hunk 2]
```diff
@@ -1550,7 +1570,7 @@ impl<'a> TryFrom<(&types::PaymentsAuthorizeRouterData, &api::WalletData)>
             additional_data,
             telephone_number: None,
             shopper_name: None,
-            shopper_email: None,
+            shopper_email: item.request.email.clone(),
```
[the remark this answers is not in the document]
Comment (prasunna09) [reply crates/router/src/connector/adyen/transformers.rs:1573]: It is only mandatory field for PayPal mandates but option for other payment methods

**Review by srujanchikke — DISMISSED** [review]

File: crates/router/src/connector/adyen/transformers.rs [hunk 3]
```diff
@@ -1521,6 +1541,24 @@ fn get_sofort_extra_details(
     }
 }
 
+fn get_shopper_email(
+    email: Option<Email>,
+    payment_method_type: storage_enums::PaymentMethodType,
+) -> errors::CustomResult<Option<Email>, errors::ConnectorError> {
+    match payment_method_type {
+        storage_enums::PaymentMethodType::Paypal => {
```
Comment (ArjunKarthik) [inline crates/router/src/connector/adyen/transformers.rs:1549]: Pass router data and use util get_email? instead of duplicating code

**Review by ArjunKarthik — APPROVED** [review]

**Review by ArjunKarthik — CHANGES_REQUESTED** [review]
One comments rest looks fine

File: crates/router/src/connector/adyen/transformers.rs [hunk 4]
```diff
@@ -1538,6 +1572,16 @@ impl<'a> TryFrom<(&types::PaymentsAuthorizeRouterData, &api::WalletData)>
         let (recurring_processing_model, store_payment_method, shopper_reference) =
             get_recurring_processing_model(item)?;
         let return_url = item.request.get_return_url()?;
+        let payment_method_type = item
+            .request
+            .payment_method_type
+            .clone()
+            .ok_or(errors::ConnectorError::MissingPaymentMethodType)?;
+        let shopper_email = if store_payment_method.is_some() {
```
Comment (ArjunKarthik) [inline crates/router/src/connector/adyen/transformers.rs:1580]: let move this condition also inside the method

**Review by ArjunKarthik — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/adyen/transformers.rs [hunk 5]
```diff
@@ -1538,6 +1577,8 @@ impl<'a> TryFrom<(&types::PaymentsAuthorizeRouterData, &api::WalletData)>
         let (recurring_processing_model, store_payment_method, shopper_reference) =
             get_recurring_processing_model(item)?;
         let return_url = item.request.get_return_url()?;
+        let shopper_email =
+            get_shopper_email(item.request.clone(), store_payment_method.is_some())?;
```
Comment (ArjunKarthik) [inline crates/router/src/connector/adyen/transformers.rs:1581]: Don't clone, pass it as reference

**Review by ArjunKarthik — CHANGES_REQUESTED** [review]

**Review by ArjunKarthik — DISMISSED** [review]

**Review by srujanchikke — DISMISSED** [review]

**Review by SanchithHegde — DISMISSED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Review by srujanchikke — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Pull request merged**